src/sempy_labs/lakehouse/_get_lakehouse_tables.py

```
@@ -83,45 +86,85 @@ def get_lakehouse_tables(
         guardrail = get_directlake_guardrails_for_sku(sku_value)
-        local_path = _mount(lakehouse=lakehouse_id, workspace=workspace_id)
+        # Avoid mounting the lakehouse if is already mounted
+        if not local_path:
+            local_path = _mount(lakehouse=lakehouse_id, workspace=workspace_id)
 
         df["Files"], df["Row Groups"], df["Table Size"] = None, None, None
         if count_rows:
             df["Row Count"] = None
 
         for i, r in df.iterrows():
+            use_schema = True
```

Ruff (0.12.2)

Local variable `use_schema` is assigned to but never used

Remove assignment to unused variable `use_schema`

(F841)